lib/fetch.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+const nodeFetch = require('node-fetch');
+const { omitBy, isUndefined } = require('lodash');
+
+const omitUndefined = object => omitBy(object, isUndefined);
+
+const errorWith = async (error, url, options, response, responseText) => Object.assign(error, {
+    meta: omitUndefined({
+        request: omitUndefined({ url, method: options.method || 'GET', body: options.body }),
+        response: response && {
+            status: response.status,
+            body: responseText
+        }
+    })
+});
+
+/**
+ * @param {string} url
+ * @param {{
+ *     expectOk?: boolean
+ *     parseResponseAs?: 'json'|'text'
+ * }} [options] - extended Fetch options
+ * @returns {Promise.<Response|string|*>} - Returns Response object or parsed body value if parseResponseAs is specified
+ */
+async function fetch (url, options = {}) {
+
+    const {
+        expectOk = true,
+        parseResponseAs,
+        ...fetchOptions
+    } = options;
+
+    if (![undefined, 'json', 'text'].includes(parseResponseAs)) {
+        throw new Error('Invalid parseResponse option.');
+    }
+
+    let response;
+
+    try {
+        response = await nodeFetch(url, fetchOptions);
+
+    } catch (error) {
+        throw errorWith(error, url, options);
+    }
+
+    if (expectOk && !response.ok) {
+        const responseText = await response.text().catch(() => 'RESPONSE BODY NOT PARSABLE');
```

Comment:
When any error occurs in response parsing, there is no way how to determine what exactly went wrong. I would propagate not only string "RESPONSE BODY NOT PARSABLE" but rather an object containing parsing error. Might be useful for debugging purposes.